### PR TITLE
Don't load parts early from MEF that aren't needed

### PR DESCRIPTION
**Pull request by jasonmalinowski**
When our package loads, we register our library objects which is what drives class view and object browser. The constructor for this proactively loads a few services which are then only used if you invoked go to definition or find references through the object browser. It's a tiny cost (25ms) in one trace but each little bit counts when trying to make load times better.

### CODE REVIEW

**Review by JoeRobich — APPROVED** [review]

File: src/VisualStudio/Core/Def/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs [hunk 1]
```diff
@@ -500,7 +494,8 @@ protected override bool TryExec(Guid commandGroup, uint commandId)
                                 // asynchronously added to the FindReferences window as they are computed.  The user
                                 // also knows something is happening as the window, with the progress-banner will pop up
                                 // immediately.
-                                _ = FindReferencesAsync(_streamingPresenter, symbolListItem, project);
+                                var streamingPresenter = ComponentModel.GetService<IStreamingFindUsagesPresenter>();
+                                _ = FindReferencesAsync(streamingPresenter, symbolListItem, project);
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs:498]: ❗ The life asynchronous operations must be captured at the call site, either by awaiting or by using a listener. We haven't figured out a way to analyze this cleanly, so it has to be seen in a review.

**Review by sharwell — CHANGES_REQUESTED** [review]

File: src/VisualStudio/Core/Def/Library/ObjectBrowser/ObjectList.cs [hunk 2]
```diff
@@ -747,8 +747,12 @@ protected override async Task GoToSourceAsync(uint index, VSOBJGOTOSRCTYPE srcTy
         {
             try
             {
-                using var token = _manager.AsynchronousOperationListener.BeginAsyncOperation(nameof(GoToSourceAsync));
-                using var context = _manager.OperationExecutor.BeginExecute(ServicesVSResources.IntelliSense, EditorFeaturesResources.Navigating, allowCancellation: true, showProgress: false);
+
+                var asynchronousOperationListener = _manager.ComponentModel.GetService<IAsynchronousOperationListenerProvider>().GetListener(FeatureAttribute.LibraryManager);
+                var operationExecutor = _manager.ComponentModel.GetService<IUIThreadOperationExecutor>();
+
+                using var token = asynchronousOperationListener.BeginAsyncOperation(nameof(GoToSourceAsync));
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Library/ObjectBrowser/ObjectList.cs:754]: 📝 There is no asynchronous work started in this method which is not already captured by this method. It is unlikely that this token is needed here; if it's used at all it should be relocated to whichever caller started an operation that wasn't awaited.

Comment (sharwell) [reply src/VisualStudio/Core/Def/Library/ObjectBrowser/ObjectList.cs:754]: Yep, the caller is incorrect:
https://github.com/dotnet/roslyn/blob/69bdfb29ce4d027c5f5bf70f50be9bebcb5df1aa/src/VisualStudio/Core/Def/Library/AbstractObjectList.cs#L317-L318

File: src/VisualStudio/Core/Def/Library/ObjectBrowser/ObjectList.cs [hunk 3]
```diff
@@ -747,8 +747,12 @@ protected override async Task GoToSourceAsync(uint index, VSOBJGOTOSRCTYPE srcTy
         {
             try
             {
-                using var token = _manager.AsynchronousOperationListener.BeginAsyncOperation(nameof(GoToSourceAsync));
-                using var context = _manager.OperationExecutor.BeginExecute(ServicesVSResources.IntelliSense, EditorFeaturesResources.Navigating, allowCancellation: true, showProgress: false);
+
+                var asynchronousOperationListener = _manager.ComponentModel.GetService<IAsynchronousOperationListenerProvider>().GetListener(FeatureAttribute.LibraryManager);
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Library/ObjectBrowser/ObjectList.cs:751]: 📝 There are multiple locations in this feature that are incorrectly handling asynchronous operations. We shouldn't rely on each location to obtain the correct specific listener on demand.

**Comment by jasonmalinowski** [comment]
@sharwell: updated some of the places to avoid the fire-and-forgets that this PR was touching. I wouldn't be surprised if there's plenty more badness in this area of the codebase, but I don't want to hold up the original intent of this PR trying to fix unrelated things.

File: src/VisualStudio/Core/Def/Library/AbstractObjectList.cs [hunk 4]
```diff
@@ -315,7 +316,10 @@ int IVsSimpleObjectList2.GoToSource(uint index, VSOBJGOTOSRCTYPE srcType)
                 return VSConstants.E_INVALIDARG;
 
             // Fire and forget
-            _ = GoToSourceAsync(index, srcType);
+            var asynchronousOperationListener = LibraryManager.ComponentModel.GetService<IAsynchronousOperationListenerProvider>().GetListener(FeatureAttribute.LibraryManager);
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Library/AbstractObjectList.cs:319]: 💡 Can we make this a property of `LibraryManager`?

**Review by sharwell — APPROVED** [review]
Would like to see the listener refactored for easy re-use in the component, but now looks good.

**Pull request merged**